libfoundation/src/foundation-java.cpp

```diff
@@ -0,0 +1,932 @@
+/* Copyright (C) 2003-2016 LiveCode Ltd.
+
+This file is part of LiveCode.
+
+LiveCode is free software; you can redistribute it and/or modify it under
+the terms of the GNU General Public License v3 as published by the Free
+Software Foundation.
+
+LiveCode is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received a copy of the GNU General Public License
+along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
+
+#include <foundation.h>
+#include <foundation-auto.h>
+
+#include "foundation-private.h"
+
+#if defined(TARGET_PLATFORM_MACOS_X) || defined(TARGET_SUBPLATFORM_ANDROID) || defined(TARGET_PLATFORM_LINUX)
+#define TARGET_SUPPORTS_JAVA
+#endif
+
+#ifdef TARGET_SUPPORTS_JAVA
+#include <jni.h>
+#endif
+
+#if defined(TARGET_PLATFORM_LINUX) || defined(TARGET_PLATFORM_MACOS_X)
+#include <stdlib.h>
+#include <dlfcn.h>
+#endif
+
+#ifdef TARGET_SUBPLATFORM_ANDROID
+extern JNIEnv *MCJavaGetThreadEnv();
+extern JNIEnv *MCJavaAttachCurrentThread();
+extern void MCJavaDetachCurrentThread();
+#endif
+
+#ifdef TARGET_SUPPORTS_JAVA
+static JNIEnv *s_env;
+static JavaVM *s_jvm;
+#endif
+
+#if defined(TARGET_PLATFORM_LINUX) || defined(TARGET_PLATFORM_MACOS_X)
+
+static bool s_weak_link_jvm = false;
+
+extern "C" int initialise_weak_link_jvm_with_path(const char*);
+static bool initialise_weak_link_jvm()
+{
+    if (s_weak_link_jvm)
+        return true;
+
+	const char *t_target;
+#if defined(TARGET_PLATFORM_LINUX)
+#if defined(__X86_64__)
+	t_target = "/jre/lib/amd64/server/libjvm.so";
+#elif defined(__I386__)
+	t_target = "/jre/lib/i386/server/libjvm.so";
```

Comment:
Unfortunately this won't even work on most Linux installations, including my Fedora VM. 😞  `getenv()` for `JAVA_HOME` any good?